src/store/diagnostics.ts

```diff
@@ -97,12 +117,23 @@ export const actions = {
 
     rowToUpdate.mute = isMuted;
 
+    const mutedChecks = { ...rootState.preferences.preferences.diagnostics.mutedChecks, [rowToUpdate.id]: isMuted };
+
+    await dispatch(
+      'preferences/commitPreferences',
+      {
+        ...rootState.credentials.credentials as ServerState,
+        payload: { diagnostics: { mutedChecks } } as Partial<Settings>,
```

Comment:
Huh, wouldn't that need to be `RecursivePartial` instead of `Partial`? (`Partial` means it expects all children to be fully there.)